src/pages-and-resources/discussions/Discussions.jsx

```diff
@@ -0,0 +1,150 @@
+import React, {
+  useCallback, useEffect, useRef, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import {
+  Switch,
+  useLocation,
+  useRouteMatch,
+} from 'react-router';
+import { history } from '@edx/frontend-platform';
+import { PageRoute } from '@edx/frontend-platform/react';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, StatefulButton } from '@edx/paragon';
+import { Check } from '@edx/paragon/icons';
+
+import FullScreenModal from '../../generic/full-screen-modal/FullScreenModal';
+import Stepper from '../../generic/stepper/Stepper';
+
+import messages from './messages';
+import AppList from './AppList';
+import ConfigFormContainer from './ConfigFormContainer';
+import { fetchApps, saveAppConfig } from './data/thunks';
+import StepperFooter from '../../generic/stepper/StepperFooter';
+import StepperHeader from '../../generic/stepper/StepperHeader';
+import StepperBody from '../../generic/stepper/StepperBody';
+import FullScreenModalHeader from '../../generic/full-screen-modal/FullScreenModalHeader';
+import FullScreenModalBody from '../../generic/full-screen-modal/FullScreenModalBody';
+
+function Discussions({ courseId, intl }) {
+  const discussionsPath = `/course/${courseId}/pages-and-resources/discussions`;
+
+  const [selectedAppId, setSelectedAppId] = useState(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const formRef = useRef();
+
+  const { path } = useRouteMatch();
+  const { pathname } = useLocation();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchApps(courseId));
+  }, [courseId]);
+
+  const onSelectApp = useCallback((appId) => {
+    if (selectedAppId === appId) {
+      setSelectedAppId(null);
+    } else {
+      setSelectedAppId(appId);
+    }
+  }, [selectedAppId]);
+
+  const onClose = useCallback(() => {
+    history.push(`/course/${courseId}/pages-and-resources`);
```

Comment:
Is there a React version of Django's `reverse` function?
Just wondering if there's a way to avoid hard-coding the URL structure..

Reply:
There isn't that I know about; routes don't have useful identifiers that you could use to reverse.  I'd like to create some helper functions to build these URLs rather than hard-coding them.